### ISSUE TITLE
crashing on inject

### ISSUE THREAD

**Issue by nikkacs** [KisSsArt/CS2-Cheat-Base#7]
as title states..

**Comment by gr1ndy** [comment]
#4 #2 already reported. wait for fix.

**Comment by asm00009** [comment]
__forceinline SchemaSystemTypeScope* findTypeScopeForModule(const char* module)
    {
        using FindTypeScopeForModule_t = SchemaSystemTypeScope* (__thiscall*)(void*, const char*);
        return VirtualMethod::getMethod<FindTypeScopeForModule_t>(this, 13)(this, module);
    }

this offset is outdated or smth

**Comment by kWAYTV** [comment]
> ```
> __forceinline SchemaSystemTypeScope* findTypeScopeForModule(const char* module)
> {
>     using FindTypeScopeForModule_t = SchemaSystemTypeScope* (__thiscall*)(void*, const char*);
>     return VirtualMethod::getMethod<FindTypeScopeForModule_t>(this, 13)(this, module);
> }
> ```
> 
> this offset is outdated or smth

i've been aswell trying to fix it with no luck

**Comment by gr1ndy** [comment]
> findTypeScopeForModule

nope. https://github.com/bruhmoment21/cs2-sdk/blob/a024d82b81c62ddc62a3fc40c0a51110b27fa9a4/cs2cheat/src/sdk/source2-sdk/interfaces/cschemasystem.hpp#L39

offset/vtable index is fine. it could prolly be sum else.

**Comment by KisSsArt** [comment]
`return VirtualMethod::getMethod<SchemaSystemTypeScope* (__thiscall*)(void*, const char*, void*)>(this, 13)(this, module, nullptr);`

here's a fix for this func. but the cheat will crash due to other problems. I don't have time to do a full fix

**Comment by kWAYTV** [comment]
> `return VirtualMethod::getMethod<SchemaSystemTypeScope* (__thiscall*)(void*, const char*, void*)>(this, 13)(this, module, nullptr);`
> 
> here's a fix for this func. but the cheat will crash due to other problems. I don't have time to do a full fix

could u please tell us where to look into it

**Comment by KisSsArt** [comment]
replace findTypeScopeForModule with this
```
SchemaSystemTypeScope* findTypeScopeForModule(const char* module)
{
    return VirtualMethod::getMethod<SchemaSystemTypeScope* (__thiscall*)(void*, const char*, void*)>(this, 13)(this, module, nullptr);
}
```

**Comment by kWAYTV** [comment]
> replace findTypeScopeForModule with this
> 
> ```
> SchemaSystemTypeScope* findTypeScopeForModule(const char* module)
> {
>     return VirtualMethod::getMethod<SchemaSystemTypeScope* (__thiscall*)(void*, const char*, void*)>(this, 13)(this, module, nullptr);
> }
> ```

no i understood that, i meant about the cs2 cheat crashing upon inject

**Comment by KisSsArt** [comment]
yeah. I almost done. I can't find time to finish it (and I'm a little lazy🥲)

**Comment by HailorLEM** [comment]
> yeah. I almost done. I can't find time to finish it (and I'm a little lazy🥲)

please do. I've actually managed to launch it, but it didn't work

**Comment by gr1ndy** [comment]
![image](https://github.com/KisSsArt/CS2-Cheat-Base/assets/71841362/864d1ee9-54de-4c6e-b520-77b95da0c82a)
apparently commeting out the "for" loop allows me to inject it using loadlibrary.

**Comment by kWAYTV** [comment]
use thread hijacking and it works

> ![image](https://user-images.githubusercontent.com/71841362/271832528-864d1ee9-54de-4c6e-b520-77b95da0c82a.png) apparently commeting out the "for" loop allows me to inject it using loadlibrary.
> 
> edit: works in game, but features other than noscope crosshair wont work

**Comment by gr1ndy** [comment]
> use thread hijacking and it works
> 
> > ![image](https://user-images.githubusercontent.com/71841362/271832528-864d1ee9-54de-4c6e-b520-77b95da0c82a.png) apparently commeting out the "for" loop allows me to inject it using loadlibrary.
> > edit: works in game, but features other than noscope crosshair wont work

i uncommented the loop and tried thread hijack, it crashes on inject. i tried injecting with the loop commented, nothing works except ns crosshair

**Comment by Jopro1990** [comment]
any updates?
